src/Integration.Vsix.UnitTests/Analysis/MuteIssueCommandTests.cs

```diff
@@ -0,0 +1,277 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Design;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.VisualStudio.OLE.Interop;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.ConnectedMode;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Binding;
+using SonarLint.VisualStudio.Core.Suppressions;
+using SonarLint.VisualStudio.Core.Transition;
+using SonarLint.VisualStudio.Infrastructure.VS;
+using SonarLint.VisualStudio.Integration.Vsix.Analysis;
+using SonarLint.VisualStudio.TestInfrastructure;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.Analysis;
+
+[TestClass]
+public class MuteIssueCommandTests
+{
+    private const int VisibleAndEnabled = (int)(OLECMDF.OLECMDF_SUPPORTED | OLECMDF.OLECMDF_ENABLED);
+    private const int VisibleButDisabled = (int)(OLECMDF.OLECMDF_SUPPORTED);
+    private const int InvisibleAndDisabled = (int)(OLECMDF.OLECMDF_SUPPORTED | OLECMDF.OLECMDF_INVISIBLE);
+
+    private static readonly BindingConfiguration ConnectedModeBinding =
+        new BindingConfiguration(default, SonarLintMode.Connected, default);
+    
+    [TestMethod]
+    public void CommandRegistration()
+    {
+        var testSubject = CreateTestSubject(out _, out _, out _, out _, out _, out _);
+        
+        testSubject.CommandID.ID.Should().Be(MuteIssueCommand.CommandId);
+        testSubject.CommandID.Guid.Should().Be(MuteIssueCommand.CommandSet);
+    }
+
+    [TestMethod]
+    public void QueryStatus_NotSonarIssue_Invisible()
+    {
+        var testSubject = CreateTestSubject(out var errorListHelperMock, out _, out _, out var activeSolutionBoundTrackerMock, out _, out _);
+        var rule = It.IsAny<SonarCompositeRuleId>();
+        errorListHelperMock.Setup(x => x.TryGetRuleIdFromSelectedRow(out rule)).Returns(false);
+        activeSolutionBoundTrackerMock
+            .SetupGet(x => x.CurrentConfiguration)
+            .Returns(ConnectedModeBinding);
+        
+        ThreadHelper.SetCurrentThreadAsUIThread();
+        var oleStatus = testSubject.OleStatus;
+
+        oleStatus.Should().Be(InvisibleAndDisabled);
+        testSubject.Visible.Should().BeFalse();
+        testSubject.Enabled.Should().BeFalse();
+    }
+
+    [TestMethod] 
+    public void QueryStatus_NotSupportedIssue_Invisible()
+    {
+        var testSubject = CreateTestSubject(out var errorListHelperMock, out _, out _, out var activeSolutionBoundTrackerMock, out _, out _);
+        SonarCompositeRuleId.TryParse("csharpsquid:S333", out SonarCompositeRuleId ruleId);
```

Comment:
using csharp for not supported rule does not make sense. We plan to support it at the end.

Reply:
It's not supported as of this commit, so I just used it for now. When we support C# this test will need to be modified, but I think it's fine. If c# is implemented properly, it will fail and remind us to modify it.